sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/UnsafeRowParquetRecordReader.java

```diff
@@ -154,6 +184,45 @@ public float getProgress() throws IOException, InterruptedException {
     return (float) rowsReturned / totalRowCount;
   }
 
+  /**
+   * Returns the ColumnarBatch object that will be used for all rows returned by this reader.
+   * This object is reused. Calling this enables the vectorized reader. It is not valid to
+   * call nextKeyValue/nextBatch before calling this.
+   */
+  public ColumnarBatch resultBatch() {
+    return resultBatch(DEFAULT_OFFHEAP);
```

Comment:
The default is false so doing what you are suggesting. I think in a follow up PR, I'm going to turn this into an enum, the boolean is confusing.